lib/dma.c

```diff
@@ -195,8 +180,9 @@ dma_controller_remove_regions(dma_controller_t *dma)
     for (i = 0; i < dma->nregions; i++) {
         dma_memory_region_t *region = &dma->regions[i];
 
-        vfu_log(dma->vfu_ctx, LOG_INFO, "unmap vaddr=%p IOVA=%lx",
-               region->virt_addr, region->dma_addr);
+        vfu_log(dma->vfu_ctx, LOG_INFO, "unmap IOVA=%#lx-%#lx vaddr=%#lx-%#lx",
+                region->iova.iov_base, iov_end(&region->iova) - 1,
+                region->vaddr.iov_base, iov_end(&region->vaddr) - 1);
 
         _dma_controller_do_remove_region(dma, region);
```

Comment:
`dma_controller_remove_regions` unconditionally calls `_dma_controller_do_remove_region`, so it will still break. We should move the `if (region->vaddr.iov_base != MAP_FAILED) ...` check from `dma_controller_remove_region` inside `_dma_controller_do_remove_region`.